manifests/named_instance.pp

```diff
@@ -0,0 +1,129 @@
+# Copyright (c) 2018 The Regents of the University of Michigan.
+# All Rights Reserved. Licensed according to the terms of the Revised
+# BSD License. See LICENSE.txt for details.
+
+# A named instance
+#
+# @example
+define nebula::named_instance(
+  String            $path,
+  Integer           $uid,
+  Integer           $gid,
+  String            $fauxpaas_pubkey,
+  String            $fauxpaas_puma_config,
+  Array[String]     $users = [],
+  Array[String]     $subservices = [],
+) {
+
+  # Create the application user's group
+  group { $title:
+    ensure => 'present',
+    gid    => $gid,
+  }
+
+  # Add sudoers and passed users to the group
+  (lookup('nebula::users::sudoers').map |$user, $_data| { $user } + $users).each |$user| {
+    exec { "${user} ${title} membership":
+      unless  => "/bin/grep -q ${title}\\S*${user} /etc/group",
+      onlyif  => "/usr/bin/id ${user}",
+      command => "/sbin/usermod -aG ${title} ${user}",
+    }
+  }
+
+  # Create the application user
+  user { $title:
+    ensure  => 'present',
+    comment => 'Application User',
+    uid     => $uid,
+    gid     => $gid,
+    home    => "/var/local/${title}",
+    shell   => '/bin/bash',
+    system  => true,
+  }
+
+  # Create the application user's home directory
+  file { "/var/local/${title}":
+    ensure => 'directory',
+    mode   => '0700',
+    owner  => $uid,
+    group  => $gid,
+  }
+
+  # Install the authorized ssh key for fauxpaas
+  ssh_authorized_key { 'fauxpaas':
+    ensure => 'present',
+    user   => $title,
+    type   => 'ssh-rsa',
```

Comment:
We probably don't want to assume the key is of type ssh-rsa (e.g. it may be ssh-ecdsa)